netket/experimental/observable/renyi2/expect.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2022 The NetKet Authors - All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from functools import partial
+
+import jax.numpy as jnp
+import jax
+import netket as nk
+
+from netket.vqs import MCState, expect
+from netket.stats import statistics as mpi_statistics
+
+from .S2_operator import Renyi2EntanglementEntropy
+
+
+@expect.dispatch
+def Renyi2(vstate: MCState, op: Renyi2EntanglementEntropy):
+    if op.hilbert != vstate.hilbert:
+        raise TypeError("Hilbert spaces should match")
+
+    return Renyi2_sampling_MCState(
+        vstate._apply_fun,
+        vstate.parameters,
+        vstate.model_state,
+        vstate.samples,
+        op.subsystem,
+    )
+
+
+@partial(jax.jit, static_argnames=("afun"))
+def Renyi2_sampling_MCState(
+    afun,
+    params,
+    model_state,
+    samples,
+    subsystem,
+):
+    N = samples.shape[-1]
+    n_chains = int(samples.shape[0] / 2)
+
+    σ_η = samples[:n_chains]
+    σp_ηp = samples[n_chains:]
```

Comment:
This won't work well. Try for example to use this with `ExactSampler` and it will crash. @alleSini99 you need to reshape the samples to a 2D matrix and then just partition those

Reply:
Ok, I have also added a test for it